packages/next/client/link.tsx

```diff
@@ -43,25 +43,81 @@ function formatUrl(url: Url) {
 }
 
 type LinkProps = {
+  elRef: any,
   href: Url,
   as?: Url|undefined,
   replace?: boolean,
-  prefetch?: boolean,
   scroll?: boolean,
   shallow?: boolean,
   passHref?: boolean
   onError?: (error: Error) => void,
 }
 
+let observer: IntersectionObserver
+const listeners = new Map()
+
+function getObserver() {
+  // Return shared instance of IntersectionObserver if already created
+  if (observer) {
+    return observer
+  }
+
+  // Only create shared IntersectionObserver if supported in browser
+  if (typeof window === 'undefined' || !(window as any).IntersectionObserver) {
```

Comment:
You can put this into the module scope:

```
const IntersectionObserver = typeof window !== 'undefined ? window.IntersectionObserver : null
```

Then it can be reused in handleRef

Reply:
Good point, updated